compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -0,0 +1,284 @@
+package nodeinventorizer
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/compliance/collection/metrics"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/scanner/database"
+	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stackrox/scanner/pkg/analyzer/nodes"
+	"golang.org/x/exp/maps"
+)
+
+const (
+	backoffMultiplier = 2
```

Comment:
I'll continue with `backoff_n+1 = backoff_n * 1.5` then, thanks for the sanity check folks!

Reply:
Correction: The `backoffMultiplier` is used to multiply a time.Duration, which is only possible with `int64`. 
There are some options:
1. We could write the number of backoffs we had so far to the cache file as well to being able to have counter-based backoffs (e.g. exponential).
2. We try to calc the number of backoffs from the found backoff (e.g. modulo)
3. We stick with `*2` multiplier

Reply:
Lets make it then `var int64` and set it to `1.5`. Would that solve the problem?

Reply:
FWIW, `x*1.5` is the same as `x*3/2`, but the latter feels fine in integer arithmetic.

Reply:
Unresolved because I'm not sure how it compiles now. I try the following code in go playground and get an error

```go
// You can edit this code!
// Click here and start typing.
package main

import (
	"fmt"
	"time"
)

const factor = 1.5

func main() {
	var a time.Duration = 69
	a = time.Duration(a * factor)
	fmt.Printf("Hello, %d\n", a)
}
```

```
./prog.go:14:24: factor (untyped float constant 1.5) truncated to int64

Go build failed.
```

Reply:
And we're back at `2`. Why I proposed `1.5` - it's because many mature libraries decide that way.
See https://github.com/cenkalti/backoff/blob/a04a6fe64ffb0e3fd0816460529d300be5f252df/exponential.go#L78
Simply cargo-culting, but I think there may be some good reasons for such choice. Also, I have heard that java containers expand their size using 1.5 multiplier.

Reply:
Sorry, seems my explanation comment got lost in the saved comments. I have investigated this as well.
I am not entirely sure how this can work at the moment - it seems like the `backoffMultiplier` is casted as needed. But I do not know what that type would be to make it work.
The solution I found reads a bit clunky, because Duration is internally implemented as int64.

If we would like to multiply a `time.Duration` by a float, we could cast the Duration into an int64, which needs to be casted into float for multiplication with 1.5. 
The result then would need to be cast back into int64 to then make a duration out of it:
```
package main

import (
	"fmt"
	"time"
)

const factor = 1.5

func main() {
	var a time.Duration = 69 * time.Second
	fmt.Println(a)

	a = time.Duration(int64(float64(int64(a)) * factor))
	fmt.Println(a)
}

```

Reply:
I tried also another variant (b2), but it is similar. I was able to compact your variant as well (b1). Have a look:

```go
var factor float64 = 1.5

func main() {
	var a time.Duration = 69 * time.Second
	b1 := time.Duration(int64(float64(a) * factor))
	b2, _ := time.ParseDuration(fmt.Sprintf("%fs", a.Seconds()*factor))

	fmt.Printf("b1 = %s * %.2f = %s\n", a.String(), factor, b1.String())
	fmt.Printf("b2 = %s * %.2f = %s\n", a.String(), factor, b2.String())
}
```

I think b1 is nicer and we can use it.